File: RepositoryPattern/app/src/main/java/com/example/vuclip/repositorypattern/MainActivity.java
package com.example.vuclip.repositorypattern;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.example.vuclip.repositorypattern.dagger.DaggerAppComponent;
import com.example.vuclip.repositorypattern.dagger.MainActivityModule;
import com.example.vuclip.repositorypattern.data.PostDataRepository;
import com.example.vuclip.repositorypattern.data.PostRepository;
import com.example.vuclip.repositorypattern.model.Post;

import java.util.List;

import javax.inject.Inject;

/**
 * Created by Banty on 08/04/18.
 */
public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Inject
    PostRepository mPostRepository;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerAppComponent.builder()
                .mainActivityModule(new MainActivityModule(this))
                .build().inject(this);

        mPostRepository.getPosts(new PostDataRepository.LoadPostsCallback() {
            @Override
            public void onPostLoaded(List<Post> posts) {
                Log.d(TAG, "onPostLoaded: " + posts.size());
            }

            @Override
            public void onPostFailedToLoad() {
                Log.d(TAG, "onPostFailedToLoad: posts failed to load");
            }
        });
    }
}

File: RecyclerViewDemo/app/src/main/java/com/example/vuclip/recyclerviewdemo2/MainActivity.kt
package com.example.vuclip.recyclerviewdemo2

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.example.vuclip.recyclerviewdemo2.adapters.MainRecyclerAdapter

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mainRecyclerView = findViewById<RecyclerView>(R.id.main_recycler_view)
        mainRecyclerView.adapter = MainRecyclerAdapter(DataProvider().getData())
        mainRecyclerView.setHasFixedSize(true)
        mainRecyclerView.layoutManager = LinearLayoutManager(this)
    }
}
